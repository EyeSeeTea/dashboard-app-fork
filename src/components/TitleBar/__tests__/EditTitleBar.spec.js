import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { EditTitleBar } from '../EditTitleBar'

jest.mock('@dhis2/d2-ui-core/text-field/TextField', () => 'textfield')
jest.mock('../../ItemSelector/ItemSelector', () => 'itemselector')

describe('EditTitleBar', () => {
    const props = {
        name: 'Rainbow Dash',
        description: 'The blue one',
        onChangeTitle: Function.prototype,
        onChangeDescription: Function.prototype,
        classes: {
            section: 'section',
            titleDescription: 'titledesc',
            title: 'title',
            description: 'description',
            underline: 'underline',
            input: 'input',
            itemSelector: 'itemSelector',
        },
    }

    it('renders correctly when displayName not provided', () => {
        const tree = shallow(<EditTitleBar {...props} />)
        expect(toJson(tree)).toMatchSnapshot()
    })

    it('renders correctly when displayName is provided', () => {
        const tree = shallow(
            <EditTitleBar displayName="Regnbue Dash" {...props} />
        )
        expect(toJson(tree)).toMatchSnapshot()
    })
})
