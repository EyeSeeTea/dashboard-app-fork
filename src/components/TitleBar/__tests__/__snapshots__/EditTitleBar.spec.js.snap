// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTitleBar renders correctly when displayName is provided 1`] = `
<section
  className="section"
>
  <div
    className="titleDescription"
  >
    <InputField
      className="title"
      dataTest="dhis2-uicore-inputfield"
      label="Dashboard title"
      name="Dashboard title input"
      onChange={[Function]}
      type="text"
      value="Rainbow Dash"
    />
    <TextAreaField
      className="description"
      dataTest="dhis2-uicore-textareafield"
      label="Dashboard description"
      name="Dashboard description input"
      onChange={[Function]}
      resize="vertical"
      rows={4}
      value="The blue one"
      width="100%"
    />
  </div>
  <div
    className="itemSelector"
  >
    <itemselector />
  </div>
</section>
`;

exports[`EditTitleBar renders correctly when displayName not provided 1`] = `
<section
  className="section"
>
  <div
    className="titleDescription"
  >
    <InputField
      className="title"
      dataTest="dhis2-uicore-inputfield"
      label="Dashboard title"
      name="Dashboard title input"
      onChange={[Function]}
      type="text"
      value="Rainbow Dash"
    />
    <TextAreaField
      className="description"
      dataTest="dhis2-uicore-textareafield"
      label="Dashboard description"
      name="Dashboard description input"
      onChange={[Function]}
      resize="vertical"
      rows={4}
      value="The blue one"
      width="100%"
    />
  </div>
  <div
    className="itemSelector"
  >
    <itemselector />
  </div>
</section>
`;
